refactor(core): always create new renderer when applying HMR metadata update

clydin opened the pull request:
The DOM renderer classes perform initialization that captures state from the component definition during construction. To ensure that the state is kept synchronized with any newly applied metadata from an HMR `applyMetadata` call, each renderer is now recreated during the apply process. This also allows inline component styles to be updated in cases where external component stylesheets may not be viable.

crisbeto commented on goldens/public-api/core/index.api.md:
```diff
@@ -1540,7 +1540,7 @@ export abstract class Renderer2 {
 export abstract class RendererFactory2 {
     abstract begin?(): void;
-    abstract createRenderer(hostElement: any, type: RendererType2 | null): Renderer2;
+    abstract createRenderer(hostElement: any, type: RendererType2 | null, force?: boolean): Renderer2;
```
This feel a bit fishy to me. Couldn't we destroy the old renderer and recreate afterwards?

clydin replied:
Update to reflect the alternative technique for clearing the renderer cache.

clydin committed:
refactor(core): always create new renderer when applying HMR metadata update

The DOM renderer classes perform initialization that captures state from
the component definition during construction. To ensure that the state is
kept synchronized with any newly applied metadata from an HMR `applyMetadata`
call, each renderer is now recreated during the apply process. This also
allows inline component styles to be updated in cases where external component
stylesheets may not be viable.

thePunderWoman commented:
This PR was merged into the repository by commit d076c3205977b958d7af9bc8ce566fc87f0fa4a3.

The changes were merged into the following branches: main, 19.0.x
